daffodil-lib/src/test/scala/passera/test/TestULong.scala

```diff
@@ -64,5 +63,45 @@ class TestULong {
     val remainder = mm1 % mm2
     assertEquals(ULong(1), remainder)
   }
-  
+
+  @Test def testULongMostNegativeLong1: Unit = {
+    val v = ULong(Long.MinValue)
+    val vbi = v.toBigInt
+    val vhex = vbi.toString(16)
+    assertEquals("8000000000000000", vhex)
+    assertEquals("8000000000000000", v.toHexString)
+    assertEquals(-9223372036854775808L, v.longValue)
+    assertEquals("9223372036854775808", v.toString)
+  }
+
+  @Test def testULongMaxValue: Unit = {
+    val v = ULong.MaxValue
+    assertEquals("FFFFFFFFFFFFFFFF", v.toHexString.toUpperCase)
+    assertEquals(ULong(0), v + ULong(1))
+    val v1 = ULong.MaxValue.toBigInt.add(JBigInt.TWO) // 0x8000000000000001
+    assertEquals(ULong(1), ULong(v1)) // preserves only 64 bits
+  }
+
+  @Test def testULongFromBigInt: Unit = {
+    val zero = JBigInt.ZERO
+    val one = JBigInt.ONE
+    val two = JBigInt.TWO
```

Comment:
Looks like JBigInt.TWO wasn't added until Java 9. I guess we'll need ``JBigInt.valueOf(2)`` or something.

Reply:
Ah, ok. Pretty easy to just do new JBigInt(2).